spectator-ext-ipc/src/main/java/com/netflix/spectator/ipc/IpcAttempt.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2014-2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spectator.ipc;
+
+import com.netflix.spectator.api.Tag;
+
+/**
+ * Dimension indicating the retry attempt.
+ *
+ * @see IpcTagKey#attempt
+ * @see IpcTagKey#attemptFinal
+ */
+public enum IpcAttempt implements Tag {
+  /** Initial request. */
+  initial,
+
+  /** Second attempt or the first retry. */
+  second,
+
+  /** A third or later attempt. */
+  third_up,
+
+  /** The attempt number cannot be determined. */
+  unknown;
+
+  @Override public String key() {
+    return IpcTagKey.attempt.key();
+  }
+
+  @Override public String value() {
+    return name();
+  }
+
+  /**
+   * Get the appropriate tag value based on the current attempt number. The attempt number
+   * should start at 1 for the initial request and increment for each retry.
+   *
+   * @param attempt
+   *     Current attempt number.
+   * @return
+   *     Tag value corresponding to the attempt number.
+   */
+  public static IpcAttempt forAttemptNumber(int attempt) {
```

Comment:
@kerumai You also have a version that accepts `Number`, is that actually used in practice? Since something like a floating point value for the attempt doesn't really make sense I would prefer the stronger typing, but we can add an overload if needed.